### PR TITLE
added support for fec mode to interfaces plugin

### PR DESCRIPTION
**Pull request by surprise30**

### CODE REVIEW

File: pkg/features/interfaces/collector.go [hunk 1]
```diff
@@ -53,6 +53,7 @@ type description struct {
 	receiveTotalErrorsDesc      *prometheus.Desc
 	transmitTotalErrorsDesc     *prometheus.Desc
 	mtu                         *prometheus.Desc
+	fecMode                     *prometheus.Desc
```
Comment (taktv6) [inline pkg/features/interfaces/collector.go]: Should be `fecModeDesc`. Maybe fix the above too.

File: pkg/features/interfaces/collector.go [hunk 2]
```diff
@@ -98,7 +99,8 @@ func newDescriptions(dynLabels dynamiclabels.Labels) *description {
 	d.receiveTotalErrorsDesc = prometheus.NewDesc(prefix+"receive_total_errors", "Number of received Total Errors", l, nil)
 	d.transmitTotalErrorsDesc = prometheus.NewDesc(prefix+"transmit_total_errors", "Number of transmitted Total Errors", l, nil)
 	d.mtu = prometheus.NewDesc(prefix+"mtu", "configured MTU", l, nil)
-
+	d.transmitTotalErrorsDesc = prometheus.NewDesc(prefix+"transmit_total_errors", "Number of transmitted Total Errors", l, nil)
```
Comment (taktv6) [inline pkg/features/interfaces/collector.go]: Duplicate of line 100?

File: pkg/features/interfaces/collector.go [hunk 3]
```diff
@@ -98,7 +99,8 @@ func newDescriptions(dynLabels dynamiclabels.Labels) *description {
...
+	d.fecMode = prometheus.NewDesc(prefix+"fec_mode", "Mode of FEC. 0 for none, 1 for fec74, 2 for fec91, 3 for fec108, 4 for unknown", l, nil)
```
Comment (taktv6) [inline pkg/features/interfaces/collector.go]: Not sure about 4 = unknown. What if a new one gets introduced? Will be 5 then?
How about 0 = off, 1 = unknown, 2 = fec74, 3 = fec91, 4 = fec108?

File: pkg/features/interfaces/collector.go (hunk 2 above)
Comment (surprise30) [reply pkg/features/interfaces/collector.go]: Done

File: pkg/features/interfaces/collector.go (hunk 1 above)
Comment (surprise30) [reply pkg/features/interfaces/collector.go]: Done

File: pkg/features/interfaces/collector.go (hunk 3 above)
Comment (surprise30) [reply pkg/features/interfaces/collector.go]: Done

**Pull request merged**